src/utils.test.ts

```diff
@@ -0,0 +1,75 @@
+import { describe, expect, test } from 'vitest';
+import { getLatestRelease, getNewVersion, isPrerelease } from './utils.js';
+
+describe('isPrerelease', () => {
+  const cases: [string, boolean][] = [
+    ['1.0.0', false],
+    ['1.0.0-0', true],
+  ];
+
+  test.each(cases)('when tag is %s', (tag: string, expectation: boolean) => {
+    expect(isPrerelease(tag)).toBe(expectation);
+  });
+});
+
+describe('getNewVersion', () => {
+  // 0: last tag
+  // 1: conventional recommended bump type
+  // 2: prerelease
+  // 3: last prod tag
+  // 4: expected release type
+  const cases: [string | null, string, boolean, string | null, string][] = [
+    ['1.0.0', 'patch', false, '1.0.0', '1.0.1'],
+    ['1.0.0', 'patch', true, '1.0.0', '1.0.1-0'],
+    ['1.0.1-0', 'patch', true, '1.0.1', '1.0.1-1'],
+    ['1.0.0-0', 'patch', true, '1.0.0', '1.0.0-1'],
+    ['1.0.1-0', 'minor', true, '1.0.1', '1.1.0-0'],
```

Comment:
**question**: if this case were to run twice using the output of the other would it result in `1.2.0-0` or `1.1.0-1`? Do we have a test case for this? Wasn't one of our issues with run away prerelease versioning?

Reply:
This test was to check against this type of thing. I added [another test](5327203c48e6307c5f223105fa8d8dde7a819ee4) to reflect your question just be sure.